js/gallery/shared/Gallery.svelte

```diff
@@ -175,27 +161,56 @@
 
 	let client_height = 0;
 	let window_height = 0;
+
+	// Unlike `gr.Image()`, images specified via remote URLs are not cached in the server
+	// and their remote URLs are directly passed to the client as `value[].image.url`.
+	// The `download` attribute of the <a> tag doesn't work with remote URLs on at least Chrome (https://stackoverflow.com/q/49474775/13103190),
+	// so we need to download the image via JS as below.
+	async function download(file_url: string, name: string): Promise<void> {
+		const response = await fetch(file_url);
+		const blob = await response.blob();
+		const url = URL.createObjectURL(blob);
+		const link = document.createElement("a");
+		link.href = url;
+		link.download = name;
+		link.click();
+		URL.revokeObjectURL(url);
+	}
+
+	$: selected_data =
+		selected_index != null && resolved_value != null
+			? resolved_value[selected_index]
+			: null;
 </script>
 
 <svelte:window bind:innerHeight={window_height} />
 
 {#if show_label}
 	<BlockLabel {show_label} Icon={Image} label={label || "Gallery"} />
 {/if}
-{#if value === null || _value === null || _value.length === 0}
+{#if value == null || resolved_value == null || resolved_value.length === 0}
 	<Empty unpadded_box={true} size="large"><Image /></Empty>
 {:else}
-	{#if selected_index !== null && allow_preview}
+	{#if selected_data && allow_preview}
 		<button on:keydown={on_keydown} class="preview">
 			<div class="icon-buttons">
 				{#if show_download_button}
-					<a
-						href={getHrefValue(value[selected_index])}
-						target={window.__is_colab__ ? "_blank" : null}
-						download="image"
-					>
-						<IconButton Icon={Download} label={i18n("common.download")} />
-					</a>
+					<div class="download-button-container">
+						<IconButton
+							Icon={Download}
+							label={i18n("common.download")}
+							on:click={() => {
+								const selected_image = selected_data?.image;
+								if (selected_image == null) {
+									return;
+								}
+								const { url, orig_name } = selected_image;
```

Comment:
With the current server-side code, `orig_name` is never set and the downloaded file name is always "image".
@abidlabs Should Python-side be modified to set `orig_name` as well?